### PR TITLE
📝 Support versioning on website

### PR DESCRIPTION
**Pull request by bisgaard-itis**
## What do these changes do?
- Support documentation of different client versions on the website. 
- Automatically generate markdown from the jupyter notebooks so that the documentation is always up-to-date and tested.

Now the front page looks like this:
image
and by hovering the mouse of the `Python` icon on the top right a dropdown appears and the user can select which client version she wants to see the documentation for. Let me know if you don't like the text I put on the front page @elisabettai. I can also show you how the documentation works if you like and then you can modify it if you want.

### CODE REVIEW

File: clients/python/Makefile [hunk 1]
```diff
@@ -86,6 +86,10 @@ install-dev: _check_venv_active ## install packages for development
 install-test: _check_venv_active install-dev ## install packages for testing client
 	pip install -r $(PYTHON_DIR)/requirements/test.txt
 
+.PHONY: install-doc
+install-test: _check_venv_active install-dev ## install packages for generating documentation
```
Comment (pcrespov) [inline clients/python/Makefile:90]: `install-test` or `install-doc`? i think the latter, right?

**Review by pcrespov — APPROVED** [review]
multi-version doc! fancy!  :-)

perhaps the main-page needs a bit more of love :-)  ... I think it should be more prominent the access to the clients e.g. a big button in the center or something. we should ask for help to  UI/UX team :-) 

In any case, just go ahead and we will iterate with that part later!

File: docs/doc_entrypoint.md [hunk 2]
```diff
@@ -1,10 +1,3 @@
...
 
-- API version: 0.4.0
-- Package version: 0.5.0
-
-## Clients
-
-Currently there is only a single client:
-
-- [Python](../clients/python/docs/README.md)
+[Osparc](https://osparc.io/) is a platform for developing, running and sharing life sience simulations. This site documents clients for interacting with osparc programatically. Choose your favorite programming language in the top-right navigation bar and get started.
```
Comment (elisabettai) [inline docs/doc_entrypoint.md:3]: I would replace the link with:

[o²S²PARC](https://sparc.science/resources/4LkLiH5s4FV0LVJd3htsvH).

And re-use the text we have in the information page.
"is an open, cloud-based platform for the development, execution and sharing of computational models, simulations, and data analysis pipelines, and the presentation of results"

**Review by elisabettai — APPROVED** [review]
Very nice! I agree with @pcrespov that the web-page looks a bit sad from the EI/UX perspective, but we can change that later. 😉 

Just a small thought: I guess we're using this website to show the BasicTutorial (and the ones that will follow in the future).

Just a small thought: we also have our pages on pypi: https://pypi.org/project/osparc/0.5.0/ for different versions. Maybe it might make sense to link them here as well?

File: clients/python/Makefile (hunk 1 above)
Comment (bisgaard-itis) [reply clients/python/Makefile:90]: Good point. Thanks.
I have corrected this

**Comment by bisgaard-itis** [comment]
> multi-version doc! fancy! :-)
> 
> perhaps the main-page needs a bit more of love :-) ... I think it should be more prominent the access to the clients e.g. a big button in the center or something. we should ask for help to UI/UX team :-)
> 
> In any case, just go ahead and we will iterate with that part later!

I definitely agree this site could need some love from a UI/UX guru.

File: docs/doc_entrypoint.md (hunk 2 above)
Comment (bisgaard-itis) [reply docs/doc_entrypoint.md]: Thanks for pointing this out. I have fixed this.

**Comment by bisgaard-itis** [comment]
> Very nice! I agree with @pcrespov that the web-page looks a bit sad from the EI/UX perspective, but we can change that later. wink
> 
> Just a small thought: I guess we're using this website to show the BasicTutorial (and the ones that will follow in the future).
> 
> Just a small thought: we also have our pages on pypi: https://pypi.org/project/osparc/0.5.0/ for different versions. Maybe it might make sense to link them here as well?


Yes, the tutorials will all be available through this site. In the current version, when one hits the version button one gets a list of tutorials which are relevant for that version (so currently only the single basic tutorial). 

About the pages on pypi I don't know exactly how that is update. I guess it is done once when we publish there. So I will definitely take this into account when we publish the next version of our package. For maintainability I guess the easiest setup would be if we just link to our "own" site there. So that we don't have to manually keep things in sync.

**Pull request merged**